keanu-project/src/test/java/io/improbable/keanu/vertices/bool/BoolVertexTest.java

```diff
@@ -95,6 +94,36 @@ public void constantVertexWorksAsExpected() {
         assertEquals(priorProbabilityTrue(a, 10000, random), p, 0.01);
     }
 
+    @Test
+    public void canObserveArrayOfValues() {
+        BoolVertex flip = new Flip(0.5);
+        boolean[] observation = new boolean[]{true, false, true};
+        flip.observe(observation);
+        assertArrayEquals(new Boolean[]{true, false, true}, flip.getValue().asFlatArray());
+    }
+
+    @Test
+    public void canSetAndCascadeArrayOfValues() {
+        BoolVertex flip = new Flip(0.5);
+        boolean[] values = new boolean[]{true, false, true};
+        flip.setAndCascade(values);
+        assertArrayEquals(new Boolean[]{true, false, true}, flip.getValue().asFlatArray());
+    }
+
+    @Test
+    public void canSetValueArrayOfValues() {
+        BoolVertex flip = new Flip(0.5);
+        boolean[] values = new boolean[]{true, false, true};
+        flip.setValue(values);
+        assertArrayEquals(new Boolean[]{true, false, true}, flip.getValue().asFlatArray());
+    }
+
+    @Test
+    public void canSetValueAsScalarOnNonScalarVertex() {
+        BoolVertex flip = new Flip(new int[]{2, 1}, 0.5);
+        flip.setValue(true);
+        assertArrayEquals(new Boolean[]{true}, flip.getValue().asFlatArray());
+    }
 
```

Comment:
add a canSetAndCascadeAsScalarOnNonScalarVertex method (as in the DoubleVertexTest)?